pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchConfig.java

```diff
@@ -286,6 +286,19 @@ public class ElasticSearchConfig implements Serializable {
     )
     private CompatibilityMode compatibilityMode = CompatibilityMode.AUTO;
 
+    @FieldDoc(
+            defaultValue = "false",
+            help = "If canonicalKeyFields is true and record key schema is JSON or AVRO, the serialized object will " +
+                    "not consider the properties order."
+    )
+    private boolean canonicalKeyFields = false;
+
+    @FieldDoc(
+            defaultValue = "null",
+            help = "Hashing algorithm to use for the document id. This is useful in order to be complaint with the ElasticSearch _id hard limit of 512 bytes."
+    )
+    private KeyHashingAlgorithm keyHashingAlgorithm;
```

Comment:
This is about the 'id' of the document. I would expect to see this name contain 'id' and not 'key'.
What about 'idGenerationAlgortithm' ?

Also I would prefer to have a NONE value because sometimes it is hard to deal with empty valued.

Reply:
Agreed
I prefer `idHashingAlgorithm` since it is not generated, it is hashed